Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -109,7 +121,18 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                         {
                             try
                             {
-                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName -ErrorAction Ignore
+                                if ($publicConfigStorageKey -eq $null)
+                                {
+                                    #if the storage account is not the classic one, get the primary key in another way
+                                    $publicConfigStorageKey = (Get-AzureRmStorageAccountKey -ResourceGroupName "$storageAccountResourceGroup" -Name "$publicConfigStorageAccountName").Value[0]
```

Comment:
Hi @Ajay-MS  thanks for your explaination. 
For the "AzurePowerShell" task, I found even if I choose the "Azure Connection Type" as "Azure classic" and also the specific Azure classic endpoint, I can still run some ARM cmdlet like "Remove-AzureRmResourceGroup" successfully.

For "AzureCloudPowerShellDeployment" task, do you mean we should not use the ARM storage account for Azure Cloud Service Deployment?  

> But Azure cloud service itself is a classic resource. So, there will be no ARM cmdlet corresponding to it. So ARM endpoint could be used here.

Currenlty we can only choose Classic Subscription from the task UI, could you please let me 
know how to use the ARM endpoint here?

Thanks.